tap_aggregator/src/main.rs

```diff
@@ -61,9 +65,18 @@ async fn main() -> Result<()> {
     tokio::spawn(metrics::run_server(args.metrics_port));
 
     // Create a wallet from the mnemonic.
-    let wallet = MnemonicBuilder::<English>::default()
-        .phrase(args.mnemonic.as_str())
-        .build()?;
+    let wallet = if let Some(key_derive_path) = args.key_derive_path.as_deref() {
+        info!("Creating wallet from mnemonic and key derive path...");
+        MnemonicBuilder::<English>::default()
+            .phrase(args.mnemonic.as_str())
+            .derivation_path(key_derive_path)?
+            .build()?
+    } else {
+        info!("Creating wallet from mnemonic...");
+        MnemonicBuilder::<English>::default()
+            .phrase(args.mnemonic.as_str())
+            .build()?
+    };
 
     // Start the JSON-RPC server.
     // This await is non-blocking
```

Comment:
To match the previous suggestion (and makes the code much simpler)
```suggestion
    tokio::spawn(metrics::run_server(args.metrics_port));

    // Create a wallet from the mnemonic.
    let wallet = MnemonicBuilder::<English>::default()
        .phrase(args.mnemonic.as_str())
        .derivation_path(&args.key_derive_path)?
        .build()?;

    info!("Wallet address: {:#40x}", wallet.address());

    // Start the JSON-RPC server.
    // This await is non-blocking
```